### PR TITLE
set timeout for bucket HEAD client

### PR DESCRIPTION
**Pull request by BenTheElder**
depends on #206 

I haven't seen evidence that this is a problem yet, but it seems obviously problematic by inspection.

There is no default timeout on http.Client. We made a similar change for the listener.

It's possible to do more granular timeouts over e.g. dialing the connection, and other tuning, but this seems like the most important and simplest change.

/hold

Not sure if this counts as a bugfix ref: #181 as currently in practice we have very good response times and haven't seen this issue yet ... but it seems bound to happen.

### CODE REVIEW

**Review by TerryHowe — APPROVED** [review]
/lgtm

File: cmd/archeio/internal/app/buckets.go [hunk 1]
```diff
@@ -91,6 +92,15 @@ func newCachedBlobChecker() *cachedBlobChecker {
...
 			cache: make(map[string]map[string]struct{}),
 		},
+		client: newHTTPClient(),
+	}
+}
+
+func newHTTPClient() *http.Client {
+	// ensure sensible timeouts
+	// this client will be used to make HEAD calls
+	return &http.Client{
```
Comment (BenTheElder) [inline cmd/archeio/internal/app/buckets.go:102]: TODO: transport and connection reuse tuning, given the large amount of calls to same-host?

Comment (BenTheElder) [reply cmd/archeio/internal/app/buckets.go:102]: we do, and that will implicitly use http.DefaultTransport
http client transport can be tuned for connection idle + reuse

Comment (BenTheElder) [reply cmd/archeio/internal/app/buckets.go:102]: client reuse may actually be suboptimal (locks on shared resources like cookies etc) vs just ensuring transport reuse and tuning that. 

Most advice about client re-use seems to be either related to transport reuse (which can instead be accomplished by passing a transport when constructing a client) or maybe allocation (not a primary concern here).

**Comment by BenTheElder** [comment]
rebased
/hold cancel

**Pull request merged**